### PR TITLE
Reimplemented switch statement for finger indices from bones

### PR DESCRIPTION
**Pull request by lucas-vrtech**

### CODE REVIEW

File: LucidGloves/src/Bones.cpp [hunk 1]
```diff
@@ -167,40 +167,61 @@ vr::HmdVector4_t CalculatePosition(const float transform, const int boneIndex, c
 	return result;
 }
 
-void ComputeBoneTransform(vr::VRBoneTransform_t bone_transform[NUM_BONES], const float transform, const int startBoneIndex, const bool isRightHand) {
+//Transform should be between 0-1
+void ComputeBoneFlexion(vr::VRBoneTransform_t *bone_transform, const float transform, int i, const bool isRightHand) {
```
Comment (danwillm) [inline LucidGloves/src/Bones.cpp:171]: Can we have the same argument names as what is in the header?

File: LucidGloves/src/Bones.cpp [hunk 2]
```diff
@@ -167,40 +167,61 @@ vr::HmdVector4_t CalculatePosition(const float transform, const int boneIndex, c
...
-		return HandSkeletonBone::eBone_Aux_Thumb;
-	case HandSkeletonBone::eBone_IndexFinger0:
-		return HandSkeletonBone::eBone_Aux_IndexFinger;
-	case HandSkeletonBone::eBone_MiddleFinger0:
-		return HandSkeletonBone::eBone_Aux_MiddleFinger;
-	case HandSkeletonBone::eBone_RingFinger0:
-		return HandSkeletonBone::eBone_Aux_RingFinger;
-	case HandSkeletonBone::eBone_PinkyFinger0:
-		return HandSkeletonBone::eBone_Aux_PinkyFinger;
+int FingerFromBone(vr::BoneIndex_t bone) {
```
Comment (danwillm) [inline LucidGloves/src/Bones.cpp:186]: is this returning indexes in the reverse order?

File: LucidGloves/src/ControllerDriver.cpp [hunk 3]
```diff
@@ -87,10 +87,11 @@ void ControllerDriver::StartDevice() {
 
 			try {
 				//Compute each finger transform
-				int currentBoneIndex = 2;
-				for (int i = 0; i < datas.flexion.size(); i++) {
-					ComputeBoneTransform(m_handTransforms, datas.flexion[i], currentBoneIndex, IsRightHand());
-					currentBoneIndex = currentBoneIndex == 2 ? currentBoneIndex += 4 : currentBoneIndex += 5;
+				for (int i = 0; i < NUM_BONES; i++) {
```
Comment (danwillm) [inline LucidGloves/src/ControllerDriver.cpp:90]: maybe this loop should be in `Bones.cpp`?

**Review by danwillm — CHANGES_REQUESTED** [review]

File: LucidGloves/src/Bones.cpp (hunk 1 above)
Comment (lucas-vrtech) [reply LucidGloves/src/Bones.cpp:171]: Sure

File: LucidGloves/src/Bones.cpp (hunk 2 above)
Comment (lucas-vrtech) [reply LucidGloves/src/Bones.cpp:186]: Depends on what you consider reverse order

File: LucidGloves/src/ControllerDriver.cpp (hunk 3 above)
Comment (lucas-vrtech) [reply LucidGloves/src/ControllerDriver.cpp:90]: You mean like in a ComputeEntireHand function? Or ComputeEntireFinger?

Comment (danwillm) [reply LucidGloves/src/ControllerDriver.cpp:90]: In fact, I think it's probably fine to have this loop there, but let's continue thinking about it.

**Review by danwillm — APPROVED** [review]
LGTM

**Pull request merged**